README.md

```diff
@@ -1 +1,248 @@
+
+```html
+<ngc-omnibox ng-model="myCtrl.model" source="myCtrl.getSuggestions(query)">
+  <ngc-omnibox-field></ngc-omnibox-field>
+
+  <ngc-omnibox-suggestions>
+    <ngc-omnibox-suggestion-item>{{suggestion.title}}</ngc-omnibox-suggestion-item>
+
+    <ngc-omnibox-suggestion-loading>Loading...</ngc-omnibox-suggestion-loading>
+    <ngc-omnibox-suggestion-empty>No results...</ngc-omnibox-suggestion-empty>
+  </ngc-omnibox-suggestions>
+</ngc-omnibox>
+```
+
+A more complicated implementation that supports multiple choices and suggestions broken down by
+category:
+
+```html
+<ngc-omnibox ng-model="myCtrl.model" source="myCtrl.getSuggestions(query)" multiple="true">
+
+  <ngc-omnibox-choices>
+    <span>
+      {{choice.title}}
+      <button ng-click="omnibox.unchoose(choice)">&times;</button>
+    </span>
+  </ngc-omnibox-choices>
+
+  <ngc-omnibox-field>
+    <input type="text" autofocus="true" placeholder="Enter your search...">
+  </ngc-omnibox-field>
+
+  <ngc-omnibox-suggestions>
+    <ngc-omnibox-category>
+      <ngc-omnibox-suggestion-header>
+        Category: {{suggestion.title}}
+      </ngc-omnibox-suggestion-header>
+
+      <ngc-omnibox-sugestion-item>
+        {{suggestion.title}} - {{suggestion.subtitle}}
+      <ngc-omnibox-suggestion-item>
+    </ngc-omnibox-category>
+
+    <ngc-omnibox-suggestion-loading>Loading...</ngc-omnibox-suggestion-loading>
+    <ngc-omnibox-suggestion-empty>No results...</ngc-omnibox-suggestion-empty>
+  </ngc-omnibox-suggestions>
+
+</ngc-omnibox>
+```
+
+For more complicated and realistic use cases, including some that implement CSS, check out the
+[examples](examples/).
+
+## Component Architecture
+
+The way to configure the markup for the Omnibox is by use of its subcomponents. The Omnibox library
+does not have its own template, so all of the markup that is displayed is completely under your
+control. Each subcomponent provides you a slot for custom markup and access to your data.
+
+### Omnibox `<ngc-omnibox>`
+
+This is the main, base-level component. All of the other subcomponents must be
+included inside of this element. Since the component does not provide a template of its own, most of
+the markup you provide inside (with some slight exceptions) will be maintained. All subcomponents
+of the Omnibox have access to the `omnibox` object in the scope of their markup, which is a
+reference to the top-level `Omnibox Controller`.
+
+### Omnibox Field `<ngc-omnibox-field>` _(Required)_
+
+The Omnibox field controls the position of the input field markup inside the Omnibox component. If
+you want to customize the actual input field element, you add an `<input>` inside this component.
+This allows you to change attributes such as `type`, `placeholder`, etc. as well as add a css class.
+
+### Omnibox Suggestions `<ngc-omnibox-suggestions>` _(Required)_
+
+Omnibox Suggestions is a container for where your list of suggestions will be rendered. It has
```

Comment:
for your lists of suggestions to be rendered in

i'm cool with ending sentences on prepositions if they sound better